core/services/pg/stats.go

```diff
@@ -41,3 +43,68 @@ func publishStats(stats sql.DBStats) {
 	promDBWaitCount.Set(float64(stats.WaitCount))
 	promDBWaitDuration.Set(stats.WaitDuration.Seconds())
 }
+
+type StatsReporterOpt func(*StatsReporter)
+
+func StatsInterval(d time.Duration) StatsReporterOpt {
+	return func(r *StatsReporter) {
+		r.interval = d
+	}
+}
+
+type DBStater interface {
+	Stats() sql.DBStats
+}
+
+type StatsReporter struct {
+	db       DBStater
+	interval time.Duration
+	cancel   context.CancelFunc
+	once     sync.Once
+}
+
+func NewStatsReporter(db DBStater, opts ...StatsReporterOpt) *StatsReporter {
+	r := &StatsReporter{
+		db:       db,
+		interval: dbStatsInternal,
+		once:     sync.Once{},
```

Comment:
nit/ no need to set zero value explicitly
```suggestion
```

Reply:
fixed